zerver/lib/markdown/fenced_code.py

```diff
@@ -103,7 +107,7 @@
     (
         \\{?\\.?
         (?P<lang>
-            [a-zA-Z0-9_+-./#]*
+            [a-zA-Z0-9_+-./# a-zA-Z0-9_+-./#]*
```

Comment:
There is no point in duplicating all the characters in this character class, and adding space will break spoiler headers.